src/extension/popup/PopupWindow.re

```diff
@@ -33,20 +85,52 @@ module PopupWindow = {
     };
     None
   };
-  let render {props, state, updater, handler} =>
+  let handleInLangChange {props, state} e => {
+    let inLanguage =
+      e |> ReactEventRe.Synthetic.target |> LocalDom.Element.value |> languageSettingOfString;
+    props.onInputChanged
+      inLang::(stringOfLanguageSetting inLanguage)
+      outLang::(stringOfLanguageSetting state.outLanguage)
+      props.inText;
+    Some {...state, inLanguage}
+  };
+  let handleOutLangChange {props, state} e => {
+    let outLanguage =
+      e |> ReactEventRe.Synthetic.target |> LocalDom.Element.value |> languageSettingOfString;
+    props.onInputChanged
+      inLang::(stringOfLanguageSetting state.inLanguage)
+      outLang::(stringOfLanguageSetting outLanguage)
+      props.inText;
+    Some {...state, outLanguage}
+  };
+  let handleInputChange {props, state} input =>
+    props.onInputChanged
+      inLang::(stringOfLanguageSetting state.inLanguage)
+      outLang::(stringOfLanguageSetting state.outLanguage)
+      input;
+  let render {props, state, updater, handler} => {
+    Js.log props;
     <div style=PopupStyles.popup>
       <div style=PopupStyles.popupColumn>
-        <h1 style=PopupStyles.popupContext> <ColumnTitle name="In" lang=props.inLang /> </h1>
+        <h1 style=PopupStyles.popupContext>
+          <ColumnTitle
+            lang=props.inLang
+            select=(select "in" (updater handleInLangChange) state.inLanguage props.inLang)
```

Comment:
It's not very clear here that `select` is a higher order component. I think I'd prefer `select` to be a barebones JSX component wrapped in a lambda.